src/containers/NetworkPanel/index.tsx

```diff
@@ -26,12 +27,21 @@ const filterNetworkRequests = (
     return networkRequests
   }
 
-  const regex = RegexParser(filterValue)
```

Comment:
I think we could abstract the use regex method out of the filterNetworkRequests function to keep the logic contained. 

It would also be possible to avoid needing to set state by changing the return value of filterNetworkRequests. This method could just return an error if the filtering was not possible. What do you think?

```ts
const getRegex = (str: string) => {
  try {
    const regex = RegexParser(str)
    return { regex, error: null }
  } catch (error) {
    return { regex: null, error }
  }
}

const filterNetworkRequests = (
  networkRequests: NetworkRequest[],
  filterValue: string,
  options: {
    isInverted: boolean
    isRegex: boolean
  }
): { results: NetworkRequest[], error?: string } => {
  if (!filterValue?.trim()?.length) {
    return { results: networkRequests }
  }

  const regexResult = options.isRegex ? getRegex(filterValue) : null
  if (regexResult?.error) {
    return { results: [], error: 'Invalid Regex' }
  }

  const results = networkRequests.filter((networkRequest) => {
    const { operationName = "" } = networkRequest.request.primaryOperation
    const isMatch = options.isRegex
      ? operationName.match(regexResult?.regex as RegExp)
      : operationName.toLowerCase().includes(filterValue.toLowerCase())

    return options.isInverted ? !isMatch : isMatch
  })

  return { results }
}
```

Then in the component, you would have access to the error without needing state/reset logic.

```ts
const { results: filterResults, error: filterError } = filterNetworkRequests(
  networkRequests,
  filterValue,
  { isInverted, isRegex: isRegexActive }
)
```

Reply:
This is such a great suggestion... so much cleaner! Thank you, I have updated the PR with your suggested changes. Let me know if there's anything else I should update! Also thanks a lot for making this awesome tool, I use it every day :D

Reply:
Thanks for making the changes. I'm really glad you're getting use from the tool :)